hail/python/test/hailtop/aiotools/test_into_copy.py

```diff
@@ -0,0 +1,41 @@
+import pytest
+import os.path
+from hailtop.aiotools.copy import copy_from_dict
+import tempfile
+
+
+def write_file(path, data):
+    with open(path, "w") as f:
+        f.write(data)
+
+
+def read_file(path):
+    with open(path, "r") as f:
+        return f.read()
+
+
+@pytest.mark.asyncio
+async def test_copy_file():
+    with tempfile.TemporaryDirectory() as test_dir:
+        write_file(f"{test_dir}/file1", "hello world\n")
+
+        inputs = [{"from": f"{test_dir}/file1", "to": f"{test_dir}/file2"},
+                  {"from": f"{test_dir}/file1", "into": f"{test_dir}/dir1"}]
+       
+        await copy_from_dict(files=inputs)
+
+        expected_files = [f"{test_dir}/file1", f"{test_dir}/file2", f"{test_dir}/dir1/file1"]
+        for file in expected_files:
+            assert read_file(file) == "hello world\n"
+
+
+@pytest.mark.asyncio
+async def test_copy_dir():
+    with tempfile.TemporaryDirectory() as test_dir:
+        os.makedirs(f"{test_dir}/subdir1")
+        write_file(f"{test_dir}/subdir1/file1", "hello world\n")
+
+        inputs = [{"from": f"{test_dir}/subdir1", "into": f"{test_dir}/subdir2"}]
+
+        await copy_from_dict(files=inputs)
+        assert read_file(f"{test_dir}/subdir2/subdir1/file1") == "hello world\n"
```

Comment:
I have another test I would like to add. If there are two files:

```
(base) # ls -l 
total 0
-rw-r--r--  1 dking  wheel  0 Jan 14 16:04 bar
-rw-r--r--  1 dking  wheel  0 Jan 14 16:04 foo
```

And we try to copy bar *into* foo, then it must fail because foo is already a file. It cannot also be a directory. Let's add a this test which:
1. creates two files 
2. tries to copy one file *into* the other file
3. and expects a `NotADirectoryError`

The pattern for "expect an error" tests is this:
```python3
try:
    # ... try to do the thing that causes an error here ...
    dangerous_function()
except TheErrorYouExpect:
    pass
else:
    assert False  # we come to "else:" if there was no error
```

Reply:
Added additional test